### PR TITLE
FM2-177:Implement chaining for search on Location.partOf

### PR DESCRIPTION
**Pull request by gitcliff**
LINK:https://issues.openmrs.org/browse/FM2-177

### CODE REVIEW

File: api/src/main/java/org/openmrs/module/fhir2/providers/r4/LocationFhirResourceProvider.java [hunk 1]
```diff
@@ -78,7 +78,9 @@ public Bundle searchLocations(@OptionalParam(name = Location.SP_NAME) StringAndL
 	        @OptionalParam(name = Location.SP_ADDRESS_POSTALCODE) StringAndListParam postalCode,
 	        @OptionalParam(name = Location.SP_ADDRESS_STATE) StringAndListParam state,
 	        @OptionalParam(name = "_tag") TokenAndListParam tag,
-	        @OptionalParam(name = Location.SP_PARTOF) ReferenceAndListParam parent, @Sort SortSpec sort) {
+	        @OptionalParam(name = Location.SP_PARTOF,chainWhitelist = { "",Location.SP_NAME, Location.SP_ADDRESS_CITY,
```
Comment (ibacher) [inline api/src/main/java/org/openmrs/module/fhir2/providers/r4/LocationFhirResourceProvider.java:81]: Should be a space after the first comma

**Comment by ibacher** [comment]
@gitcliff This looks good so far! We do need some additional things done:

1. This same change should be implemented for the [R3 version of the LocationFhirResourceProvider](https://github.com/openmrs/openmrs-module-fhir2/blob/master/api/src/main/java/org/openmrs/module/fhir2/providers/r3/LocationFhirResourceProvider.java).
2. We need to add appropriate tests for the [LocationFhirResourceProviderTest](https://github.com/openmrs/openmrs-module-fhir2/blob/master/api/src/test/java/org/openmrs/module/fhir2/providers/r4/LocationFhirResourceProviderTest.java#L53) class

**Comment by gitcliff** [comment]
@ibacher have added the tests and added the change in R3 to

File: api/src/main/java/org/openmrs/module/fhir2/providers/r4/LocationFhirResourceProvider.java (hunk 1 above)
Comment (gitcliff) [reply api/src/main/java/org/openmrs/module/fhir2/providers/r4/LocationFhirResourceProvider.java]: sure,,done

**Comment by ibacher** [comment]
@gitcliff Thanks for this! Could you also implement that test for the R3 provider? Also be sure to run `mvn clean package` before committing. Your import order looks somewhat different from the order we enforce for this package and will cause issues down the line. Thanks!

**Comment by gitcliff** [comment]
@ibacher its done

**Pull request merged**